File: templates/app/skeleton/backend/gofiber/internal/handler/health.go
package handler

import (
	"os"

	"github.com/gofiber/fiber/v2"
	"gorm.io/gorm"
)

// HealthHandler provides the health check endpoint.
type HealthHandler struct {
	DB *gorm.DB
}

// Health godoc
// @Summary     Health check
// @Tags        ops
// @Produce     json
// @Success     200 {object} map[string]string
// @Failure     503 {object} map[string]string
// @Router      /healthz [get]
func (h *HealthHandler) Health(c *fiber.Ctx) error {
	if os.Getenv("DATABASE_URL") != "" && h.DB == nil {
		return c.Status(fiber.StatusServiceUnavailable).JSON(fiber.Map{
			"status": "unhealthy",
			"reason": "database not connected",
		})
	}
	return c.JSON(fiber.Map{"status": "ok"})
}

File: templates/app/skeleton/backend/fastapi/app/handler/health.py
"""DB-aware health check endpoint."""

import os

from fastapi import APIRouter, Response

from app.model.db import get_engine

router = APIRouter()


@router.get("/healthz", tags=["ops"])
def health(response: Response) -> dict[str, str]:
    """Health check — returns 503 if DATABASE_URL is set but DB is not connected."""
    if os.getenv("DATABASE_URL") and get_engine() is None:
        response.status_code = 503
        return {"status": "unhealthy", "reason": "database not connected"}
    return {"status": "ok"}
